### ISSUE TITLE
Use with `pipeline` or `generate`

### ISSUE THREAD

**Issue by helleuch** [tomaarsen/attention_sinks#7]
Hello,
Thank you very much for making this work available. I would like to ask you if this works with the `transformer.pipeline` or the `model.generate` ? Or do we have to use a `TextStreamer` as per the example ?

**Comment by tomaarsen** [comment]
`TextStreamer` is a very simple class from transformers that only has one goal: printing out tokens as they are being generated - it's kind of separate from the generation process itself.

You can do generation like `model.generate` just like in normal `transformers`. 
Only if you want to use multi-step generation (i.e. where the previous outputs are added as history to new prompts), then you should use a [manual generation loop](https://github.com/tomaarsen/attention_sinks/blob/f46e63101fa74c6095e986c33284217c34a9fd88/demo/streaming.py#L47-L55) that calls `model(input_ids, use_cache=True, past_key_values=past_key_values)` and extracts the `past_key_values` for the next inputs and extracts the logits to determine the generated token.

- Tom Aarsen

**Comment by helleuch** [comment]
Thank you very muck !
